### PR TITLE
Merge `release-0.4.0-dev` into `main`

### PR DESCRIPTION
**Pull request by tchughesiv**

### CODE REVIEW

File: config/manifests/kustomization.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ patchesJson6902:
     # Remove the manager container's "cert" volumeMount, since OLM will create and mount a set of certs.
     # Update the indices in this path if adding or removing containers/volumeMounts in the manager's Deployment.
     - op: remove
-      path: /spec/template/spec/containers/1/volumeMounts/0
+      path: /spec/template/spec/containers/0/volumeMounts/0
```
Comment (redhatHameed) [inline config/manifests/kustomization.yaml:23]: @tchughesiv Encountered an error while building the bundle locally. Just wondering if this only fix for local builds ? as per comments OLM will mount the path automatically.

File: controllers/dbaasplatform_controller.go [hunk 2]
```diff
@@ -135,7 +137,7 @@ func (r *DBaaSPlatformReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	}
 
 	defer func() {
-		metrics.SetPlatformMetrics(*cr, cr.ClusterName, execution, event, metricLabelErrCdValue)
+		metrics.SetPlatformMetrics(*cr, cr.Name, execution, event, metricLabelErrCdValue)
```
Comment (redhatHameed) [inline controllers/dbaasplatform_controller.go:140]: @rhkp we have change this, as were facing compilation issue `ClusterName`. let's know if required any fix here.

File: config/manifests/kustomization.yaml (hunk 1 above)
Comment (tchughesiv) [reply config/manifests/kustomization.yaml:23]: this is required due to the sdk/kustomize version upgrades. they swap the position of the manager container thus the array # change.

Comment (redhatHameed) [reply config/manifests/kustomization.yaml:23]: Got it, thanks

File: controllers/dbaasplatform_controller.go (hunk 2 above)
Comment (redhatHameed) [reply controllers/dbaasplatform_controller.go:140]: @rhkp thanks, for fixing this into main branch on this PR merged.

**Review by redhatHameed — APPROVED** [review]

**Review by jeremyary — APPROVED** [review]

**Pull request merged**